File: python/HelloWorld_main.py
#!/usr/bin/env python3
"""Print 'Hello World!' and exit with code 0 on success or 1 on error."""

import sys


def hello_world() -> None:
    """Print 'Hello World!' to standard output."""
    print("Hello World!")


def main() -> int:
    """Execute the main logic. Return 0 on success, 1 on failure."""
    try:
        hello_world()
    except OSError as e:
        print(f"ERROR: {e}", file=sys.stderr)
        return 1
    return 0


if __name__ == "__main__":
    sys.exit(main())

File: ruby/HelloWorld_main.rb
#!/usr/bin/env ruby
# frozen_string_literal: true

# Print 'Hello World!' to standard output.
def hello_world
  puts 'Hello World!'
end

# Execute the main logic. Return 0 on success, 1 on failure.
def main
  hello_world
  0
rescue SystemCallError => e
  warn "ERROR: #{e}"
  1
end

if __FILE__ == $PROGRAM_NAME
  exit(main)
end
